DragaliaAPI/Features/Login/LoginBonusService.cs

```diff
@@ -36,17 +33,28 @@ public async Task<IEnumerable<AtgenLoginBonusList>> RewardLoginBonus()
             )
         )
         {
-            int currentDay = (int)(time - bonusData.StartTime).TotalDays;
-            int dayId = bonusData.IsLoop
-                ? currentDay
-                    % MasterAsset.LoginBonusReward.Enumerable.Count(x => x.Gid == bonusData.Id)
-                : currentDay;
+            DbLoginBonus dbBonus = await loginBonusRepository.Get(bonusData.Id);
+            if (dbBonus.IsComplete)
+            {
+                logger.LogDebug(
+                    "Player has already completed login bonus {@bonus}, skipping...",
+                    dbBonus
+                );
+                continue;
+            }
 
-            dayId += 1;
+            dbBonus.CurrentDay += 1;
```

Comment:
this addition cannot preceed the modulo operation as it is meant to prevent dayId == 0 which can still happen here

Reply:
moved